wagtail/contrib/modeladmin/helpers/permission.py

```diff
@@ -28,6 +28,13 @@ def get_all_model_permissions(self):
             content_type__model=self.opts.model_name,
         )
 
+    def get_all_permission_codenames(self):
```

Comment:
Suggestion: This might be better implemented as a `cached_property` rather than manually caching. 

```python
@cached_property
def all_permission_codenames(self):
    ...
```